ops/model.py

```diff
@@ -117,7 +119,11 @@ def get_unit(self, unit_name: str) -> 'Unit':
         Internally this uses a cache, so asking for the same unit two times will
         return the same object.
         """
-        return self._cache.get(Unit, unit_name)
+        if unit_name == self._backend.unit_name:
+            container_names = tuple(self._meta.containers)
+        else:
+            container_names = ()
+        return self._cache.get(Unit, unit_name, container_names)
```

Comment:
This looks suspect. The primary key of the unit is indeed its name. It's been a while since I went through that code, but do we really want containers as part of the constructor, rather than passing them into that single unit somehow?

Reply:
Yeah, good call. I knew it wasn't great when I was adding it, but didn't think of a better way. I've cleaned this up now so the constructors/key is only the name again, and the "meta" is attached to the `_ModelCache` instance, which things already have access to. This is clean, and the cache seems like a reasonable place to hang it off of.